### ISSUE TITLE
Export not working

### ISSUE THREAD

**Issue by Boulouboup** [OGRECave/blender2ogre#246]
Hi I'm trying to export meshes into .mesh files but it's not working. I'm currently using **Blender 3.6.12** and **blender2ogre 0.9.0** on Windows 10. Here are my settings:

![OgreToolsPath](https://github.com/user-attachments/assets/2485fdb6-1cbf-49a9-9917-bb706ade01de)

When I'm trying to export, I have the following messages:

![OgreXMLConverter problem](https://github.com/user-attachments/assets/56922353-2679-4dfc-b43c-c1f6af28a79a)
![OgreXMLConverter_problem](https://github.com/user-attachments/assets/e58d36d4-7b73-4788-bbc7-d96777ae4953)

There seems to be an error with OgreXMLConverter but I don't know how to solve it. I used the latest Ogre SDK.

Here are the logs:

![blender2ogre_log](https://github.com/user-attachments/assets/0a2a1fba-32f0-49b1-baad-1a53d94350d8)
![OgreXMLConverter_log](https://github.com/user-attachments/assets/041f2a96-3b43-420d-b3bb-63d807b420fd)

It worked fine with blender2ogre 0.8.3 but I will want to make my addon compatible with Blender 4.X.X also.

Here is how I call the blender2ogre export operator:

![blender2ogre_call](https://github.com/user-attachments/assets/8ab88a9f-af1f-4d4e-90da-ab9c841c139f)

Thanks in advance for your help!

**Comment by Boulouboup** [comment]
It works on simpler mesh. It fails when I try to export more complex mesh objects that I imported from **step** file with an addon. But as I said it worked with blender2ogre 0.8.3 so I don't think the problem is the step file.

**Comment by sercero** [comment]
Hello, can you share the model?

**Comment by sercero** [comment]
> `Error: Missing <colour_diffuse> element`

The problem might be there in the OgreXMLConverter log output so it seems that the `<colour_diffuse>` component is not being exported for some reason.

**Comment by Boulouboup** [comment]
Hi, the model I'm using is a step file that I imported with STEP OCC Import which is a paid addon, I don't know if you have it so i'm sharing the blend file:

[export_fail.zip](https://github.com/user-attachments/files/16437783/export_fail.zip)

**Comment by Boulouboup** [comment]
I just tried with blender2ogre 0.8.3 and it's working fine, the export worked. I'ma stick with 0.8.3 for now.

**Comment by sercero** [comment]
@Boulouboup the issue is with the name of the `Colour Atribute`, the name is `Colours` and we are expecting a name from this list: `color_names = ["col", "color", "colour", "attribute"]` (case insensitive)

In previous versions we had this:
```
            if self.__colors is None and self.__alphas is None:
                # No alpha and color found by name, assume that the only
                # vertex color data is actual color data
                self.__colors = colors
```

But it was removed.

Can you try changing the color names in the attribute to `Color` and trying again?

I'm curious if there are other issues before pushing a fix

**Comment by Boulouboup** [comment]
> Can you try changing the color names in the attribute to `Color` and trying again?

Sorry I'm not an expert, where exactly do I have to change the attribute, in the **mesh.py** file? Are you saying I should try to modifiy blender2ogre code directly?

**Comment by sercero** [comment]
> > Can you try changing the color names in the attribute to `Color` and trying again?
> Sorry I'm not an expert, where exactly do I have to change the attribute, in the **mesh.py** file? Are you saying I should try to modifiy blender2ogre code directly?

No, you modify that in your model. Select each object then go to vertex (the three green triangles) and then expand "color attributes" and then change the name to "color" instead of "colors" do that for every mesh in your model. And then try to export again.
 
> By the way do you guys know how to get an installed addon **version** with python, like get the version of blender2ogre in blender? I can't find anything working on google.

You can see it in the console when Blender start, do you need to get it programatically?

**Comment by Boulouboup** [comment]
> No, you modify that in your model. Select each object then go to vertex (the three green triangles) and then expand "color attributes" and then change the name to "color" instead of "colors" do that for every mesh in your model. And then try to export again.

Ok I'll check that, thanks for you help.

> You can see it in the console when Blender start, do you need to get it programatically?

Yes programmatically

**Comment by Boulouboup** [comment]
> No, you modify that in your model. Select each object then go to vertex (the three green triangles) and then expand "color attributes" and then change the name to "color" instead of "colors" do that for every mesh in your model. And then try to export again.

It's working, thank you!! How are you so good??

**Comment by sercero** [comment]
> > No, you modify that in your model. Select each object then go to vertex (the three green triangles) and then expand "color attributes" and then change the name to "color" instead of "colors" do that for every mesh in your model. And then try to export again.
> 
> It's working, thank you!! How are you so good??

Well, I'm the one who introduced the bug in the first place 😅🤦‍♂️🙈

So, I'll put that piece of code back to catch the situations where the name is not on the list.

**Comment by sercero** [comment]
Leave this issue Open so that I reference it in a pull request.

**Comment by Boulouboup** [comment]
By the way, it has nothing to do with issues or bugs but, I'm a beginner in python and blender, and I'm coding my addon in VS Code with separated modules, each time I'm modifying something and want to try it, I have to uninstall my addon on Blender, close Blender, launch it, reinstall my addon etc... it's very long and annoying, I wanted to know how you guys are writing your scripts and how you test it. I'm using Blender 3.6.12, I can't find anything helpful on the internet, I heard about that VS Code extension "Blender Development" but I can't make it work.

**Comment by sercero** [comment]
I'm not using it, but there is the fake BPY module:
https://github.com/nutti/fake-bpy-module

What it does is simulate a Blender environment for you to develop in python.

Regarding the reloading of the addon you have to look for the ability to reload the modules
https://blender.stackexchange.com/questions/2691/is-there-a-way-to-restart-a-modified-addon

Basically in Blender by pressing F8 you can reload the scripts/addons, but the scripts need to be prepared for that.

**Comment by sercero** [comment]
Solved by #247
